device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/exceptions/MultiplexingClientDeviceRegistrationFailedException.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.sdk.iot.device.exceptions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Exception that is thrown when one or more devices fail to register to an active multiplexed connection.
+ */
+public class MultiplexingClientDeviceRegistrationFailedException extends MultiplexingClientException
+{
+    private Map<String, Exception> registrationExceptions = new HashMap<>();
+
+    /**
+     * Construct a new MultiplexingClientDeviceRegistrationFailedException with no nested exception and no error message.
+     */
+    public MultiplexingClientDeviceRegistrationFailedException()
+    {
+        super();
+    }
+
+    /**
+     * Construct a new MultiplexingClientDeviceRegistrationFailedException with no nested exception but with an error message.
+     * @param message The top level message for this exception.
+     */
+    public MultiplexingClientDeviceRegistrationFailedException(String message)
```

Comment:
I should maybe rename this to ```MultiplexingClientDeviceRegistrationAuthenticationFailedException``` to be more specific, since it is only ever thrown when a device's authentication fails

Reply:
I think that is a good change. you probably don't even need the word `failed` in there as Exception kind of gives that away